front50-s3/src/main/java/com/netflix/spinnaker/front50/config/S3Config.java

```diff
@@ -0,0 +1,81 @@
+package com.netflix.spinnaker.front50.config;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.awsobjectmapper.AmazonObjectMapper;
+import com.netflix.spinnaker.clouddriver.aws.bastion.BastionConfig;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.front50.model.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.web.client.RestTemplate;
+import rx.schedulers.Schedulers;
+
+import java.util.concurrent.Executors;
+
+@Configuration
+@Import(BastionConfig.class)
+public class S3Config {
+  @Value("${spinnaker.s3.bucket}")
+  private String bucket;
+
+  @Value("${spinnaker.s3.rootFolder}")
+  private String rootFolder;
+
+  @Bean
+  public AmazonClientProvider amazonClientProvider() {
+    return new AmazonClientProvider();
+  }
+
+  @Bean
+  public AmazonS3 awsS3Client(AWSCredentialsProvider awsCredentialsProvider) {
+    return new AmazonS3Client(awsCredentialsProvider);
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(RestTemplate.class)
+  public RestTemplate restTemplate() {
+    return new RestTemplate();
+  }
+
+  @Bean
+  public ObjectMapper amazonObjectMapper() {
+    return new AmazonObjectMapper();
+  }
+
+  @Bean
+  @ConditionalOnExpression("${spinnaker.s3.enabled:false}")
```

Comment:
I see your point. Because it is s3, I was viewing that as a subsystem. Like
redis and cassandra. Unlike those, s3 isnt using the option, spinnaker is.
Also, in the presence of others, like gs or other platform variants, I'd
think you'd want them grouped/related together rather than in the global
space.
So the grouping category doesnt need to be 'spinnaker'. It could be
something else like dao. I dont have a feel for the big picture. I was
reacting to abstract policy more than the particular details.

On Tue, Mar 15, 2016 at 11:12 AM, Adam Jordens <notifications@github.com>
wrote:

> In
> front50-s3/src/main/java/com/netflix/spinnaker/front50/config/S3Config.java
> <https://github.com/spinnaker/front50/pull/73#discussion_r56179546>:
>
> > +    return new AmazonS3Client(awsCredentialsProvider);
> > +  }
> > +
> > +  @Bean
> > +  @ConditionalOnMissingBean(RestTemplate.class)
> > +  public RestTemplate restTemplate() {
> > +    return new RestTemplate();
> > +  }
> > +
> > +  @Bean
> > +  public ObjectMapper amazonObjectMapper() {
> > +    return new AmazonObjectMapper();
> > +  }
> > +
> > +  @Bean
> > +  @ConditionalOnExpression("${spinnaker.s3.enabled:false}")
>
> The issue I suppose is that this is the only service with scoped config,
> which looks a little strange when I look at config on a service-by-service
> basis.
>
> And even then there will be core cassandra config expected to live at the
> root/unscoped level, ie:
> cassandra.embedded.
>
> Then again, there are plenty of inconsistencies in config styles across
> services so perhaps this is no worse.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/spinnaker/front50/pull/73/files#r56179546
>